providers/base/updater.go

```diff
@@ -1,91 +0,0 @@
-package base
```

Comment:
Moving all of this logic to the base provider since this abstraction is probablly too much.